scipy/linalg/_decomp_polar.py

```diff
@@ -99,6 +99,18 @@ def polar(a, side="right"):
     a = np.asarray(a)
     if a.ndim != 2:
         raise ValueError("`a` must be a 2-D array.")
+    if a.size == 0:
+        dt = a.dtype
+        if side == 'right':
+              # a = up
+              u = np.empty_like(a, dtype = dt)
+              p = np.empty_like(np.transpose(a).dot(a), dtype = dt)
```

Comment:
Minor but would be nice to tweak, since this needs an update anyway

```suggestion
              p = np.empty_like(a.T @ a, dtype=dt)
```

(use the matmul operator for readability; no spaces around = sign for keyword args. Here and elsewhere.)